packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1808,15 +1834,32 @@ export class ContainerRuntime
 	public process(messageArg: ISequencedDocumentMessage, local: boolean) {
 		this.verifyNotClosed();
 
+		const updateSummaryHeuristics = (message: ISequencedDocumentMessage) => {
+			if (this._summarizer !== undefined && this.groupedBatchingEnabled) {
```

Comment:
I'd rather raise some event (if we do not have proper event yet) and summarizer layer to listen on these events.
Adding this direct dependency / coupling is not great